### PR TITLE
Small updates to generalize paths/prediction functions

### PR DESCRIPTION
**Pull request by themattinthehatt**

### CODE REVIEW

File: lightning_pose/utils/predictions.py [hunk 1]
```diff
@@ -118,24 +130,37 @@ def predict_single_video(
...
+    The function will need to initialize a DALI pipeline, prepare a dataloader, and pass
+    it on to _make_predictions().
 
     Args:
-        video_file (str): absolute path to a single video you want to get predictions for. typically .mp4 file.
-        ckpt_file (str): absolute path to the checkpoint of your trained model. assumed .ckpt format.
-        cfg_file (Union[str, DictConfig]): either a hydra config or a path pointing to one, with all the model specs. needed for loading the model.
-        preds_file (str): absolute filename for the predictions .csv file.
-        sequence_length (int, optional): number of frames in a sequence of video frames drawn by DALI. Defaults to 16. can be controlled externally by cfg.eval.dali_parameters.sequence_length.
+        video_file (str): absolute path to a single video you want to get predictions
```
Comment (danbider) [inline lightning_pose/utils/predictions.py:144]: nice docstring we have here

File: lightning_pose/utils/predictions.py [hunk 2]
```diff
@@ -192,40 +217,33 @@ def predict_single_video(
...
-    except (PermissionError):
+    except PermissionError:
         new_save_file = os.path.join(os.getcwd(), preds_file.split("/")[-1])
         save_dframe(df, new_save_file)
         print(
             "Couldn't save file to the desired location due to a PermissionError. Instead saved it in %s"
             % new_save_file
         )
+
+    if heatmaps_np is not None:
```
Comment (danbider) [inline lightning_pose/utils/predictions.py:235]: that's the `os.makedirs` call i was referring to

File: lightning_pose/utils/predictions.py [hunk 3]
```diff
@@ -565,9 +594,8 @@ def save_dframe(df: pd.DataFrame, save_file: str) -> None:
 
 
 @typechecked
-def save_heatmaps(heatmaps_np: np.ndarray, save_folder: str) -> None:
+def save_heatmaps(heatmaps_np: np.ndarray, save_file: str)-> None:
     import h5py
-
-    save_file = os.path.join(save_folder, "heatmaps.h5")
+    assert save_file.endswith(".h5") or save_file.endswith(".hdf5")
```
Comment (danbider) [inline lightning_pose/utils/predictions.py:599]: good addition

File: scripts/train_hydra.py [hunk 4]
```diff
@@ -127,8 +127,10 @@ def train(cfg: DictConfig):
     predict_dataset(
         cfg=cfg,
         data_module=data_module,
-        hydra_output_directory=hydra_output_directory,
         ckpt_file=model_ckpt,
+        preds_file=os.path.join(hydra_output_directory, "predictions.csv"),
+        heatmap_file=os.path.join(
+            hydra_output_directory, "heatmaps_and_images", "heatmaps.h5"),
```
Comment (danbider) [inline scripts/train_hydra.py:133]: the name `"heatmaps_and_images"` sounds outdated. all the image stuff happens later in fiftyone. we just need to save an `.h5` file with heatmaps, no?

File: scripts/train_hydra.py [hunk 5]
```diff
@@ -127,8 +127,10 @@ def train(cfg: DictConfig):
...
     )
 
     # generate a video
```
Comment (danbider) [inline scripts/train_hydra.py:136]: Also wondering if we should predict new vids here or not? can postpone that for later. But all of the users report immediately moving to video predictions.

File: tests/utils/test_df_ops.py [hunk 6]
```diff
@@ -1,11 +1,13 @@
+
 def test_dfConverter():
```
Comment (danbider) [inline tests/utils/test_df_ops.py:2]: that's my useless test here. don't mind removing

**Review by danbider — CHANGES_REQUESTED** [review]
all looks good; added a few minor suggestions. LMK what you think.

File: lightning_pose/utils/predictions.py [hunk 7]
```diff
@@ -72,44 +113,15 @@ def predict_dataset(
...
-        or save_file.endswith(".hdf")
-        or save_file.endswith(".h5")
-        or save_file.endswith(".h")
-    ):
-        raise NotImplementedError("Currently only .csv and .h5 files are supported")
+    # save heatmaps
+    if heatmaps_np is not None:
+        save_folder = os.path.dirname(heatmap_file)
+        if not (os.path.isdir(heatmap_file)):
+            os.mkdir(save_folder)
```
[the remark this answers is not in the document]
Comment (themattinthehatt) [reply lightning_pose/utils/predictions.py:123]: good catch, will change to `os.makedirs`

File: scripts/train_hydra.py (hunk 4 above)
Comment (themattinthehatt) [reply scripts/train_hydra.py:133]: agreed, I removed that extra directory

File: tests/utils/test_df_ops.py (hunk 6 above)
Comment (themattinthehatt) [reply tests/utils/test_df_ops.py:2]: we'll leave for now, maybe we can expand it later

**Pull request merged**